### ISSUE TITLE
No CSRF protection in the admin pages

### ISSUE THREAD

**Issue by northdpole** [Hackademic/hackademic#55]
The admin pages should have some protection from csrf attacks, a very simple and effective solution is to use esapi's csrf token generator to fix this.

For any would-be patchers, read first owasp's csrf protection cheat sheet

**Comment by northdpole** [comment]
assigned to @NullArgument

**Comment by mebjas** [comment]
I see this hasn't been fixed yet. I see we can easily use https://www.owasp.org/index.php/CSRFProtector_Project here. If you think its a good idea, I'd integrate this myself.

**Comment by pchaigno** [comment]
By all means, go ahead!

**Comment by mebjas** [comment]
I tested CSRF Protector for `hackademic` in the `next` branch,  It worked well! Then I realised I was supposed to send pull to `next-dev` branch where there is already some implementation for CSRF mitigation using `noCSRF`. 
**What to do?**
As a suggestion I'd say using CSRF Protector is better as it requires you to just include a library and call init method, rest it does itself. Also it provide protection for both form elements and ajax requests (both GET & POST).

P.S: I'm one of developer behind CSRFP so you shall do your own research before selecting one!

**Comment by northdpole** [comment]
@mebjas yes, please switch noCsrf to the protector when NullArgument contirbuted it, he didn't know about the protector. 
Does the csrf protector project support multipart forms? (the forms where you upload files)

When you submit a pull request, can you please also provide a few unit tests testing that your contribution works as expected? It would be awkward if someone in the future commits something that brakes csrf protection and we never find out :p

**Comment by mebjas** [comment]
yeah `CSRFP` attaches a token for every HTML Form by default if it is a POST request (for `GET` you need to configure the url).

And its cool I'd make necessary changes as required and write unit tests as well.

**Comment by islamoc** [comment]
@northdpole any updates about this ? or I can work on it ?

**Comment by northdpole** [comment]
yes you can, go for it, the branch is next-dev (although the cms is going for a rewrite this summer)
